api/base/filters.py

```diff
@@ -171,7 +188,11 @@ def param_queryset(self, query_params, default_queryset):
         if fields_dict:
             for field_name, value in fields_dict.items():
                 if self.is_filterable_field(key=field_name):
-                    queryset = queryset.intersection(set(self.get_filtered_queryset(field_name, value, default_queryset)))
+                    filtered_queryset = set(self.get_filtered_queryset(field_name, value, default_queryset))
+                    queryset = [
+                        result for result in queryset
+                        if result in filtered_queryset
+                    ]
```

Comment:
Great explanation. I totally missed the existence of line 190.

I feel like there's got to be a way where we can loop through the result set fewer times but nothing is jumping out at me right now. This works and does what it should. I'd say it's potentially ripe for optimization though. Anyone filtering by more than one field with a result set > 0 is going to be sitting in this function for a while.

Two more tiny things. 
- query_params_to_fields returns a dictionary unless an exception occurs so you could kill 189 and just let 190 loop zero times if the dict is empty.
- @abought pointed out that the end of 211 could potentially call `.lower()` on `None` which wouldn't work out well for anyone. I know that's not part of your changes but it's probably worth replacing `None` with `''` or something equally benign while you're in there.